### ISSUE TITLE
Failed to accept valid certificate (during validating kubeconfig)

### ISSUE THREAD

**Issue by jeka** [lensapp/lens#57]
While trying to add a cluster with following kubeconfig:
```
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data:   -- snip (irrelevant) -- 
    server: https://10.87.2.61/k8s/clusters/c-hjx5f
  name: sandbox
contexts:
- context:
    cluster: sandbox
    namespace: default
    user: sandbox
  name: sandbox
current-context: sandbox
kind: Config
preferences: {}
users:
- name: sandbox
  user:
    token: -- snip --
```
I am getting the following error:
```
Invalid kubeconfig context sandbox
Error [ERR_TLS_CERT_ALTNAME_INVALID]: Hostname/IP does not match certificate's altnames: IP: 10.87.2.61 is not in the cert's list:
```
Note that the cert's list is empty.
Although:
```
openssl s_client -connect 10.87.2.61:443 < /dev/null | openssl x509 -noout -text | grep DNS:
Can't use SSL_get_servername
depth=0 O = the-ranch, CN = cattle
verify error:num=20:unable to get local issuer certificate
verify return:1
depth=0 O = the-ranch, CN = cattle
verify error:num=21:unable to verify the first certificate
verify return:1
DONE
                DNS:, IP Address:10.87.2.61, IP Address:192.168.1.11
```
- Lens Version: 2.3.2
- OS: Linux
- Installation method: App Image 
 
And kubectl used from command line works with the same kubeconfig without a problem.

**Comment by jakolehm** [comment]
Is the server Rancher 2.x?

**Comment by jeka** [comment]
yes, it is Rancher 2

**Comment by jeka** [comment]
More info: tried to access k3s with the following config:
```
  apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUJXRENCL3FBREFnRUNBZ0VBTUFvR0NDcUdTTTQ5QkFNQ01DTXhJVEFmQmdOVkJBTU1HR3N6Y3kxelpYSjIKWlhJdFkyRkFNVFUzTWpJM056VTJOVEFlRncweE9URXdNamd4TlRRMk1EVmFGdzB5T1RFd01qVXhOVFEyTURWYQpNQ014SVRBZkJnTlZCQU1NR0dzemN5MXpaWEoyWlhJdFkyRkFNVFUzTWpJM056VTJOVEJaTUJNR0J5cUdTTTQ5CkFnRUdDQ3FHU000OUF3RUhBMElBQk9MMkhnNkFnZHVkMThzOFptN2VEK2xjdm9KdkFXWjBnTU03UzViMXhDTTUKZldraElQdFFjSnV5SVlWL29aYm5CTFZCYS9WaTh5UVJmdjNhRENSR0FKMmpJekFoTUE0R0ExVWREd0VCL3dRRQpBd0lDcERBUEJnTlZIUk1CQWY4RUJUQURBUUgvTUFvR0NDcUdTTTQ5QkFNQ0Ewa0FNRVlDSVFDTkYyd09RK2w1CjZiZksrS2JZQmt4cW1hd05DL2FvRDZxVWcraUJYaDUyQ3dJaEFNU0U4bGk2OThvVXJydndXaWtxcHpBNEFBOE8KVC92MFRKbFMrSHpJQUJIbAotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==
    server: https://192.168.122.210:6443
  name: default
contexts:
- context:
    cluster: default
    namespace: default
    user: default
  name: default
current-context: default
kind: Config
preferences: {}
users:
- name: default
  user:
    password: 712f65faad2d2a54f994d48062a73848
    username: admin
```
Strangely enough, this time Lens does not complain about the certificate. But I still can't add the cluster. The error is:
```
Invalid kubeconfig context default
￼ Show details
Error: tunneling socket could not be established, cause=socket hang up
```
And of course kubectl have no problem connecting with the same config.
```
 $ kubectl version
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T19:18:23Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.4-k3s.1", GitCommit:"a7531b1ab3fd5ff987b074472ddfb84a2f5326bc", GitTreeState:"clean", BuildDate:"2019-09-19T22:36Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by jakolehm** [comment]
Update: will be fixed in 2.6.0.

**Comment by jakolehm** [comment]
2.6.0 is out.
